### ISSUE TITLE
Questions about utils.py line 112, function label propagation

### ISSUE THREAD

**Issue by S-rz** [DongHande/PT_propagation_then_training#2]
`y0 = torch.zeros(size=(labels.shape[0], labels.max().item() + 1))`
I wanna ask  why here use labels.shape[0] for dimension 0, are nodes all have labels on these four datasets?  Should it more suitable to use adj.shape[0]?
Furthermore, do you know some papers for large-scale label propagation since matrix multiplication is expensive when has a lot of nodes?

**Comment by DongHande** [comment]
Thanks for your interest in our paper. 
For the first question: All nodes have labels on the four datasets. Thus, labels.shape[0] == adj.shape[0] == (the number of nodes in the graph). 
For the second question: (1) For large-scale graph, the amount of computation for any algorithm will be larger. (2) Since the matrix multiplication for label propagation is sparse matrix multiplication ( [Sparse A] * [dense Y]) and the dimensions of labels are always small, label propagation is not expensive.

**Comment by S-rz** [comment]
Thanks for your work and reply, it really inspired me a lot. Wish you all the best!
